packages/fs-detectors/test/unit.builds-and-routes-detector.test.ts

```diff
@@ -2359,6 +2359,26 @@ describe('Test `detectBuilders` with `featHandleMiss=true`', () => {
     ]);
   });
 
+  it('should not add middleware builder when building "nextjs" + undefined project settings', async () => {
+    const files = ['package.json', 'pages/index.ts', 'middleware.ts'];
+    const pkg = {
+      scripts: { build: 'next build' },
+      dependencies: { next: '12.2.0' },
+    };
+    const { builders } = await detectBuilders(files, pkg, {
+      featHandleMiss,
+    });
+    expect(builders).toEqual([
+      {
+        use: '@vercel/next',
+        src: 'package.json',
+        config: {
+          zeroConfig: true,
```

Comment:
This should never happen. This should have `framework: nextjs` here if it really is Next.js

Reply:
Hello, I copied the test directly above this one and changed the arguments that `detectBuilders()` is called with so that it will have `undefined` for `projectSettings` and, by proxy, the `framework` field in the project settings (which is what happens with the field there is no Vercel config file in a project). The above test was testing when the `framework` option had the value `null`. This is testing when it is `undefined` as this is fixing a long-time bug with that case.

The outputted builder is the exact same that the other Next.js tests uses.

Reply:
I understand the test you created, but I dont understand the conditions for which it can happen.

Do you have a deployment url where this happened?

Reply:
The issue is that you have a fake `project.json` file here: https://github.com/james-elicx/app-playground/blob/main/.vercel/project.json

If you run through the steps to create a project, then it will work correctly:

```
vc link
vc pull
vc build
```

Here are the settings I see:

```json
{
  "projectId": "prj_...",
  "orgId": "team_...",
  "settings": {
    "createdAt": 1690568775159,
    "framework": "nextjs",
    "devCommand": null,
    "installCommand": null,
    "buildCommand": null,
    "outputDirectory": null,
    "rootDirectory": null,
    "directoryListing": false,
    "nodeVersion": "18.x"
  }
}
```